functions/index.js

```diff
@@ -255,91 +308,73 @@ const updateUser = (req, res) => {
         });
 };
 
-// User Post
 
-const newPost = (req, res) => {
-    const Post = {
-        username: req.user.username,
-        post: req.body.post,
-        tags: req.body.tags,
-        canLikes: req.body.canLikes,
-        canUnlike: req.body.canUnlike,
+ 
 
+// User Post
+// funciona
+const newPost = (req, res) => {
+    const Postuser = {
+        username: req.body.username ||"",
+        post: req.body.post ||"",
+        tag: req.body.tag ||"",
+        canLikes: req.body.canLikes ||"",
+        canUnlike: req.body.canUnlike ||"",
     };
-    if (Post.post == undefined) {
+    if (Postuser.post == undefined) {
         return res.status(400).json({ error: "Missing require parameter" });
     }
+    console.log("Starting Upload...", Postuser.username);
 
-    console.log("Starting Upload...", newUser.username);
-
-    db.collections("posts")
-        .add(Post)
+    db.doc(`/posts/${Postuser.post}`)
+        .create(Postuser)
         .then((doc) => {
             return res.status(201).json({ message: "Post created successfully" });
+
         })
         .catch((error) => {
             res.status(500).json({ error: error.code });
         });
 };
 
 // Update User Post
-
+// funciona
 const updatedPost = (req, res) => {
-    const post = {
-        post: req.body.post,
-        tags: req.body.tags,
-        postId: req.body.postId,
+    const editpost = {
+        post: req.body.post ||"",
+        tags: req.body.tags ||"",
+        title: req.body.title ||"",
 
     };
-    if (post.post == undefined) {
+    const username = req.body.username;
+    if (editpost.post == undefined) {
         return res.status(400).json({ error: "Missing requirement" });
     }
 
-    console.log("Post edited", user.username);
-    db.doc(`/posts/${post.postId}`)
-        .get()
-        .then((Post) => {
-            if (!Post.exists) {
-                throw {
-                    code: "post/non-existent",
-                    error: new Error(),
-                };
-            } else {
-                const existingPost = doc.data();
-                const updatedPost = {
-                    post: post.post || existingPost.post,
-                    tags: post.tags || existingPost.tags,
-                };
-                return db.doc(`/posts/${post.postId}`).update(updatedPost);
-            }
-        })
-        .then(() => {
-            return res.satus(200).json({ message: "Post updated successfully" });
+    console.log("Post edited", username);
+    db.doc(`/posts/${editpost.post}`)
```

Comment:
if you implement what I suggested above and get rid of the editpost.post in the database entry, the way to update will be to receive a postId from the front-end as a parameter. So your new structure looks like the following:

```
const editPost = {
   postId: req.params.postId,
   post: req.body.post || "",
   tags: req.body.tags|| "",
   title: req.body.title || ""
} 
```